ssz/codec.py

```diff
@@ -4,19 +4,37 @@
 
 from ssz.exceptions import (
     DecodingError,
+    InvalidSedesError,
 )
 from ssz.utils import (
+    get_sedes_from_string,
     infer_sedes,
+    is_sedes,
 )
 
 
-def encode(obj, serializer_type=None):
+def encode(obj, sedes=None):
     """
     Encode object in SSZ format.
-    `serializer_type` needs to be explicitly mentioned for encode/decode
-    of integers(as of now)
+    `sedes` needs to be explicitly mentioned for encode/decode
+    of integers(as of now).
+    `sedes` parameter could be given as a string or as the
+    actual sedes object itself.
     """
-    serialized_obj = infer_sedes(obj, serializer_type).serialize(obj)
+    if sedes:
+        if isinstance(sedes, str):
```

Comment:
@jannikluhn but this approach wouldn't be generic enough for the new `uintN` type, for which we didn't hard code. For example, what if a user wants to give `uint24`(rare but possible), but we wouldn't hardcode that, which would lead to a `InvalidSedesError`. So how do we handle these use cases?

Reply:
We can hardcode all common values (e.g. everything between 8 and 64 as well as the values used in the spec).